### PR TITLE
#1164 fix the member_max_fails missing param

### LINKED ISSUE #1164: Upstream member:  member_max_fails param is missing
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.3.3-1
- Ruby:  -
- Distribution: Debian Jessie
- Module version: 0.9.0

## How to reproduce (e.g Puppet code you use)

Call the `nginx::resource::upstream::member` with the parameter:  `upstream_max_fails` 

## What are you seeing

Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Nginx::Resource::Upstream::Member[test]: has no parameter named 'upstream_max_fails' at 

## What behaviour did you expect instead
Setting the upstream member to the right max fails counter.

## Any additional information you'd like to impart
the parameter is already available in the member template.
it's only missing in the manifest. 
I've created a PR.

### PR DESCRIPTION
**Pull request by khaefeli**
fix #1164

### CODE REVIEW

File: manifests/resource/upstream/member.pp [hunk 1]
```diff
@@ -39,7 +40,8 @@
   $server,
   Enum['present', 'absent'] $ensure = 'present',
   Integer $port                     = 80,
-  $upstream_fail_timeout  = '10s',
+  $upstream_fail_timeout            = '10s',
```
Comment (juniorsysadmin) [inline manifests/resource/upstream/member.pp]: Given these are new, can you use data types?

Comment (khaefeli) [reply manifests/resource/upstream/member.pp]: done

**Comment by joekohlsdorf** [comment]
I don't agree with the use of String datatypes for time values, I suggest the following:
```Optional[Variant[Integer,Pattern[/^\d+[smhdwMy]?$/]]]```

**Comment by bastelfreak** [comment]
Hi @khaefeli, thanks for the PR. Can you take a look at the comment from @joekohlsdorf? Can you please add a unit test?

**Comment by saz** [comment]
I don't think this PR is required anymore: https://github.com/voxpupuli/puppet-nginx/blob/52ecfd7e5f08cf3c7046cc37a3af2fc069763732/manifests/resource/upstream/member.pp#L57

**Comment by khaefeli** [comment]
closed. not needed anymore

**Pull request closed without merging**